library/Fission/Internal/Mock/Types.hs

```diff
@@ -0,0 +1,251 @@
+import           Fission.Web.Auth.Class
+import           Fission.Models
+import Fission.IPFS.DNSLink.Class
+import           Fission.Web.Server.Reflective.Class
+
+
+import           Fission.AWS
+import qualified Fission.Platform.Heroku.Auth.Types as Heroku
+
+import           Fission.Internal.Mock.Session.Types
+import           Network.IPFS.Local.Class
+
+import           Fission.Internal.Mock.Config.Types as Mock
+
+import           Fission.User                  as User
+import           Fission.User.CID              as User.CID
+import           Fission.Platform.Heroku.AddOn as Heroku.AddOn
+
+{- | Fission's mock type
+
+     Notes:
+     * We will likely want @State@ in here at some stage
+     * @RIO@ because lots of constraints want @MonadIO@
+       * Avoid actual @IO@, or we're going to have to rework this 😉
+
+-}
+newtype Mock effs a = Mock
+  { unMock :: WriterT [OpenUnion effs] (RIO Mock.Config) a }
+  deriving
+    newtype ( Functor
+            , Applicative
+            , Monad
+            , MonadWriter [OpenUnion effs]
+            , MonadReader Mock.Config
+            , MonadIO
+            , MonadThrow
+            , MonadCatch
+            )
+
+instance IsMember RunDB effs => MonadDB (Mock effs) (Mock effs) where
+  runDB transaction = do
+    Effect.log RunDB
+    transaction
+
+instance MonadLinkedIPFS (Mock effs) where
+  getLinkedPeers = do
+    peerList <- asks linkedPeers
+    return peerList
+
+instance MonadAuth Text (Mock effs) where
+  verify = do
+    isAuthed <- asks forceAuthed
+    return <| BasicAuthCheck \_ ->
+      return <| if isAuthed
+                  then Authorized "YUP"
+                  else Unauthorized
+
+instance MonadAuth (Entity User) (Mock effs) where
+  verify = asks userVerifier
+
+instance MonadAuth Heroku.Auth (Mock effs) where
+  verify = asks herokuVerifier
+
+instance IsMember RunAWS effs => MonadAWS (Mock effs) where
+  liftAWS awsAction = do
+    Effect.log RunAWS
+    env <- newEnv <| FromKeys "FAKE_ACCESS_KEY" "FAKE_SECRET_KEY"
+
+    awsAction
+      |> runAWST env
+      |> runResourceT
+      |> liftIO
+
+instance IsMember CheckTime effs => MonadTime (Mock effs) where
+  currentTime = do
+    Effect.log CheckTime
+    asks now
+
+instance
+  ( IsMember RunAWS        effs
+  , IsMember UpdateRoute53 effs
+  )
+  => MonadRoute53 (Mock effs) where
+  update r d t = do
+    Effect.log UpdateRoute53
+    runner <- asks updateRoute53
+    return <| runner r d t
+
+instance
+  ( IsMember UpdateRoute53 effs
+  , IsMember SetDNSLink    effs
+  , IsMember RunAWS        effs
+  )
+  => MonadDNSLink (Mock effs) where
+  set s c = do
+    Effect.log SetDNSLink
+    runner <- asks setDNSLink
+    return <| runner s c
+
+instance IsMember RunLocalIPFS effs => MonadLocalIPFS (Mock effs) where
+  runLocal _ _ = do
+    Effect.log RunLocalIPFS
+    asks localIPFSCall
+
+instance IsMember RunRemoteIPFS effs => MonadRemoteIPFS (Mock effs) where
+  runRemote = undefined 
```

Comment:
Right now this class has `runRemote` as required, when the other options should be to define the other functions. Needs to be fixed in `ipfs-haskell`